mscore/macos/cocoabridge.mm

```diff
@@ -20,14 +20,43 @@
 #include "cocoabridge.h"
 #import <Cocoa/Cocoa.h>
 
-void  CocoaBridge::setAllowsAutomaticWindowTabbing(bool flag)
-{
-    if ([NSWindow respondsToSelector:@selector(allowsAutomaticWindowTabbing)])
-        [NSWindow setAllowsAutomaticWindowTabbing: flag];
-}
-
-bool  CocoaBridge::isSystemDarkTheme()
-{
-    NSString *osxMode = [[NSUserDefaults standardUserDefaults] stringForKey:@"AppleInterfaceStyle"];
-    return ([osxMode isEqualToString:@"Dark"]);
-}
+id<NSObject> darkModeObserverToken;
```

Comment:
I've never seen such a syntax. Would you mind explaining what it means?

Reply:
This is Objective-C code, which may explain why you are not familiar with the syntax. The purpose of the CocoaBridge class is to allow MuseScore to interact with the Cocoa subsystem on macOS, which is accomplished by using Objective-C. Since `cocoabridge.h` is valid C++ syntax as well as valid Objective-C syntax, the same header file can be #included by it’s own implementation file as well as any C++ sources that want to use the CocoaBridge class. If you have spent some time trying to learn Obective-C, and still have no idea what is going on, please let us know.

Reply:
I have not spent time learning Objective-C:) And I had no idea it was possible to use non-C++ classes in a c++ project. That's actually kinda mindblowing. I'll have to dig on stackoverflow...

Reply:
The macOS implementation of the Qt libraries is full of Objective-C code. This is no different. And StackOverflow is great, but it is not the first place I would turn to for a tutorial on Objective-C.

Reply:
It wasn't about a tutorial on objective-C, but rather about combining c++ and objective-C in the same codebase. IMHO, SO is not so great apart for this kind of *wild* questions.

Reply:
No need to search on StackOverflow. All you need to know is that Xcode can compile Objective-C code as well as C++ code and link them together.

Reply:
Yes, as @mattmcclinch said, it's Objective-C, which is the programming language Apple mainly uses for their frameworks. So CocoaBridge.mm is quite literally a bridge between our/Qt's C++ code and Apple's Objective-C code. 
I must admit that I don't have a very deep knowledge of Objective-C, but it is my most unfavourite language, primarily because of it's weird syntax with those square brackets. In C++, you would just write something like:
```cpp
object.method(val);
```
But with Objective-C objects, it would be:
```objc
[object method:val];
```
In my opinion, it looks bad, and it is inconvenient to type. 

If you're interested in native macOS/iOS developing, I recommend that you have a look at the Swift programming language. Apple has made this as a replacement for Objective-C, and I think it's a really nice and powerful language. 

Btw: This is the Objc-course I used: https://www.youtube.com/watch?v=5esQqZIJ83g but there are many others.